benchmarks/benchmark_base.py

```diff
@@ -0,0 +1,453 @@
+
+    def get_relative_path(self, partial_path: str):
+        path = dirname(realpath(__file__))
+        targets = self.split_path(partial_path)
+
+        for target in targets:
+            path = join(path, target)
+
+        return path
+
+    def get_absolute_path(self, partial_path):
+        partial_path = "../" + partial_path
+
+        return self.get_relative_path(partial_path)
+
+    @property
+    def tardis_config_verysimple(self):
+        filename = self.get_absolute_path("tardis/io/configuration/tests/data/tardis_configv1_verysimple.yml")
+        return yaml_load_file(
+            filename,
+            YAMLLoader,
+        )
+
+    @property
+    def tardis_config_verysimple_nlte(self):
+        filename = self.get_absolute_path("tardis/io/configuration/tests/data/tardis_configv1_nlte.yml")
+        return yaml_load_file(
+            filename,
+            YAMLLoader,
+        )
+
+    @property
+    def nlte_raw_model_root(self):
+        return SimulationState.from_config(
+            self.tardis_model_config_nlte_root, self.nlte_atom_data
+        )
+
+    @property
+    def nlte_raw_model_lu(self):
+        return SimulationState.from_config(
+            self.tardis_model_config_nlte_lu, self.nlte_atom_data
+        )
+
+    @property
+    def nlte_atom_data(self):
+        atomic_data = deepcopy(self.nlte_atomic_dataset)
+        return atomic_data
+
+    @property
+    def nlte_atomic_dataset(self):
+        nlte_atomic_data = AtomData.from_hdf(self.nlte_atomic_data_fname)
+        return nlte_atomic_data
+
+    @property
+    def nlte_atomic_data_fname(self):
+        atomic_data_fname = f"{self.tardis_ref_path}/nlte_atom_data/TestNLTE_He_Ti.h5"
+        atom_data_missing_str = f"{atomic_data_fname} atomic datafiles " f"does not seem to exist"
+
+        if not Path(atomic_data_fname).exists():
+            raise Exception(atom_data_missing_str)
+
+        return atomic_data_fname
+
+    @property
+    def tardis_ref_path(self):
+        # TODO: This route is fixed but needs to get from the arguments given in the command line.
+        #       /app/tardis-refdata
+        return '/app/tardis-refdata'
+
+    @property
+    def atomic_dataset(self) -> AtomData:
+        atomic_data = AtomData.from_hdf(self.atomic_data_fname)
+
+        if atomic_data.md5 != DEFAULT_ATOM_DATA_UUID:
+            message = f'Need default Kurucz atomic dataset (md5="{DEFAULT_ATOM_DATA_UUID}")'
+            raise Exception(message)
+        else:
+            return atomic_data
+
+    @property
+    def atomic_data_fname(self):
+        atomic_data_fname = f"{self.tardis_ref_path}/atom_data/kurucz_cd23_chianti_H_He.h5"
+
+        if not Path(atomic_data_fname).exists():
+            atom_data_missing_str = f"{atomic_data_fname} atomic datafiles " f"does not seem to exist"
+            raise Exception(atom_data_missing_str)
+
+        return atomic_data_fname
+
+    @property
+    def tardis_model_config_nlte_root(self):
```

Comment:
To be clear, I mean that the order of the properties in the class definition as written in the file would be better in a clear order. Right now they are not clearly ordered by usage or name. If it can be broken up into multiple classes, even better.

Reply:
Could you check my demo proposal in the commit: 3aa02eb8de191a226fdd768ff2e15099ddbbfa67?

If it is useful and there are no comments, I'll continue to refactor the `benchmark_base.py` and all the benchmarks with this structure.

---

The benchmarks ran as expected:

```text
· Creating environments                                                                                                             12:20:35 [8/8]
· Discovering benchmarks
· Running 12 total benchmarks (1 commits * 1 environments * 12 benchmarks)
[ 0.00%] · For tardis commit b7e07720 <master>:
[ 0.00%] ·· Benchmarking mamba-py3.12
[ 4.17%] ··· Running (io_configuration_config_reader.BenchmarkIoConfigurationConfigReader.time_config_hdf--).....
[25.00%] ··· Running (io_configuration_config_reader.BenchmarkIoConfigurationConfigReader.time_plasma_nlte_section_lu_config--).
[29.17%] ··· Running (io_configuration_config_reader.BenchmarkIoConfigurationConfigReader.time_plasma_nlte_section_root_config--).
[33.33%] ··· Running (io_configuration_config_reader.BenchmarkIoConfigurationConfigReader.time_plasma_section_config--).....
[54.17%] ··· io_configuration_config_reader.BenchmarkIoConfigurationConfigReader.time_config_hdf                                         69.4±2ms
[58.33%] ··· io_configuration_config_reader.BenchmarkIoConfigurationConfigReader.time_convergence_section_parser                      1.27±0.02μs
[62.50%] ··· io_configuration_config_reader.BenchmarkIoConfigurationConfigReader.time_from_config_dict                                   58.2±1ms
[66.67%] ··· io_configuration_config_reader.BenchmarkIoConfigurationConfigReader.time_model_section_config                               128±10ms
[70.83%] ··· io_configuration_config_reader.BenchmarkIoConfigurationConfigReader.time_plasma_nlte_root_exc_section_config               22.3±0.4s
[75.00%] ··· io_configuration_config_reader.BenchmarkIoConfigurationConfigReader.time_plasma_nlte_section_lu_config                     21.7±0.1s
[79.17%] ··· io_configuration_config_reader.BenchmarkIoConfigurationConfigReader.time_plasma_nlte_section_root_config                  21.4±0.01s
[83.33%] ··· io_configuration_config_reader.BenchmarkIoConfigurationConfigReader.time_plasma_section_config                                    ok
[83.33%] ··· ================= ============
               Plasma values
             ----------------- ------------
              initial_t_inner   60.9±0.3ms
               initial_t_rad     61.7±1ms
             ================= ============

[87.50%] ··· io_configuration_config_reader.BenchmarkIoConfigurationConfigReader.time_spectrum_section_config                            64.3±1ms
[91.67%] ··· io_configuration_config_reader.BenchmarkIoConfigurationConfigReader.time_supernova_section_config                            131±2ms
[95.83%] ··· plasma_nlte_excitation_x.BenchmarkPlasmaNlteExcitation.time_coll_exc_deexc_matrix                                                 ok
[95.83%] ··· ============================================================================================================================== =============
                                                                         Matrix
             ------------------------------------------------------------------------------------------------------------------------------ -------------
                          {'coll_exc_coeff_values': [1, -2, 3], 'coll_deexc_coeff_values': [4, 9, 10], 'number_of_levels': 3}                1.63±0.09ms
              {'coll_exc_coeff_values': [0.21, 0.045, 0.1234], 'coll_deexc_coeff_values': [0.7865, 0.987, 0.00123], 'number_of_levels': 3}   1.57±0.04ms
             ============================================================================================================================== =============

[100.00%] ··· plasma_nlte_excitation_x.BenchmarkPlasmaNlteExcitation.time_prepare_bound_bound_rate_matrix                                  140±2ms

real    5m32.771s
user    5m40.852s
sys     1m52.319s
```